src/csharp/akka_validation/Network.cs

```diff
@@ -136,7 +136,10 @@ public Subscribe(IActorRef subscriber, Type topic)
         private ImmutableDictionary<ulong, IActorRef> _idToRef = ImmutableDictionary<ulong, IActorRef>.Empty;
         private ImmutableDictionary<IActorRef, ulong> _refToId = ImmutableDictionary<IActorRef, ulong>.Empty;
         private int _nextMessageId;
-        private readonly ConcurrentDictionary<int, Task> _waitingForOmnetToDelvierMessage = new ConcurrentDictionary<int, Task>();
+        /// <summary>
+        /// The key is a tuple of MessageId and ReceiverId. After a broadcast we need to distinguish different receivers, because message are delivered at different points in time.
+        /// </summary>
+        private readonly ConcurrentDictionary<Tuple<int, ulong>, Task> _waitingForOmnetToDelvierMessage = new ConcurrentDictionary<Tuple<int, ulong>, Task>();
```

Comment:
We now store not only the message id, but also bind each task to a specific receiver id.
For a broadcast we create multiple tasks that can individually be triggered.